src/compiler/compile/render_dom/wrappers/AwaitBlock.ts

```diff
@@ -60,7 +60,7 @@ class AwaitBlockBranch extends Wrapper {
 
 			this.block.chunks.declarations.push(b`(${node.pattern} = #ctx[${index}])`);
 			if (this.block.has_update_method) {
-				this.block.chunks.update.push(b`(${node.pattern} = #ctx[${index}])`);
+				this.block.chunks.update.unshift(b`(${node.pattern} = #ctx[${index}])`);
```

Comment:
Rather than using `unshift` instead of `push` here, I prefer to reorder the `render` and `render_destructure` calls in the `AwaitBlockWrapper#render`

```diff
+		this.then.render_destructure(block, this.value, this.node.value, value_index);
+		this.catch.render_destructure(block, this.error, this.node.error, error_index);

		[this.pending, this.then, this.catch].forEach(branch => {
			branch.render(branch.block, null, x`#nodes` as Identifier);
		});

-		this.then.render_destructure(block, this.value, this.node.value, value_index);
-		this.catch.render_destructure(block, this.error, this.node.error, error_index);
```